File: patterns/java/src/refactoring_guru/patterns/visitor/example/visitor/Visitor.java
package refactoring_guru.patterns.visitor.example.visitor;

import refactoring_guru.patterns.visitor.example.shapes.*;

public interface Visitor {
    public String visitDot(Dot dot);

    public String visitCircle(Circle circle);

    public String visitRectangle(Rectangle rectangle);

    public String visitCompoundGraphic(CompoundShape cg);
}

File: patterns/cs/src/RefactoringGuru/Visitor/Example/Visitor/Visitor.cs
using System;
using RefactoingGuru.Visitor.Example.Shapes;

namespace RefactoingGuru.Visitor.Example.Visitor
{
    interface IVisitor
    {
        String VisitDot(Dot dot);

        String VisitCircle(Circle circle);

        String VisitRectangle(Rectangle rectangle);

        String VisitCompoundGraphic(CompoundShape cg);
    }
}
